src/cursor/core_cursor.ts

```diff
@@ -218,14 +205,9 @@ class CoreCursor extends Readable {
     nextFunction(this, callback);
   }
 
-  /**
-   * Clone the cursor
-   *
-   * @function
-   * @returns {CoreCursor}
-   */
+  /** Clone the cursor */
   clone(): CoreCursor {
-    return this.topology.cursor(this.ns, this.cmd, this.options);
+    return new CoreCursor(this.topology, this.operation, this.options);
```

Comment:
topology.cursor makes use of the cursorFactory option, is there any concern about that being dropped here?